pkg/utils/utils.go

```diff
@@ -56,17 +58,38 @@ func RetrieveValFromMap[K comparable, V any](m map[K]V, key K) V {
 // PollUntil validates if a certain condition is met at defined poll intervals.
 // If a timeout is reached, an associated error is returned to the caller.
 // condition contains the use-case specific code that returns true when a certain condition is achieved.
-func PollUntil(pollInterval, timeOut <-chan time.Time, condition func() (bool, error)) error {
+func PollUntil(pollInterval, timeOut <-chan time.Time, condition func() (string, bool, error)) error {
```

Comment:
can we have different function which uses spinner and leave this default one as is? may be like `SpinnerPollUntil`

Reply:
Yes, makes sense since we are changing the function's signature